### PR TITLE
Add com.rustdesk.RustDesk

### PR DESCRIPTION
**Pull request by besdar**
### Please confirm your submission meets all the criteria

- [X] Please describe the application briefly.

An open-source remote desktop application designed for self-hosting, as an alternative to TeamViewer. 

- [x] The domain used for the application ID is [controlled by the application developers][appid-domain] and the [application id guidelines][appid] are followed.
- [X] I have read the and followed all the [Submission and licence requirements][reqs].
- [X] I have [built][build] and tested the submission locally.
- [X] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about this submission. **Link:** https://github.com/rustdesk/rustdesk/discussions/1537#discussioncomment-10533094

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

File: com.rustdesk.RustDesk.json [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+      "build-commands": [
+        "bsdtar -zxvf rustdesk.deb",
+        "tar -xvf ./data.tar.xz",
+        "cp -r usr/* /app/",
+        "mkdir -p /app/bin && ln -s /app/lib/rustdesk/rustdesk /app/bin/rustdesk",
+        "mv /app/share/applications/rustdesk.desktop /app/share/applications/com.rustdesk.RustDesk.desktop",
+        "sed -i '/^Icon=/ c\\Icon=com.rustdesk.RustDesk' /app/share/applications/*.desktop",
+        "mv /app/share/icons/hicolor/scalable/apps/rustdesk.svg /app/share/icons/hicolor/scalable/apps/com.rustdesk.RustDesk.svg",
+        "install -Dm644 com.rustdesk.RustDesk.metainfo.xml /app/share/metainfo/com.rustdesk.RustDesk.metainfo.xml",
+        "for size in 16 24 32 48 64 128 256 512; do\n rsvg-convert -w $size -h $size -f png -o $size.png rustdesk.svg\n install -Dm644 $size.png /app/share/icons/hicolor/${size}x${size}/apps/com.rustdesk.RustDesk.png\n done"
```
Comment (hfiguiere) [inline com.rustdesk.RustDesk.json:64]: you don't need to convert anything. The SVG is sufficient.

File: com.rustdesk.RustDesk.desktop
Comment (hfiguiere) [inline com.rustdesk.RustDesk.desktop:1]: the .deb doesn't have a .desktop?

File: com.rustdesk.RustDesk.desktop [hunk 2]
```diff
@@ -0,0 +1,11 @@
+[Desktop Entry]
+Name=RustDesk
+GenericName=RustDesk
+Comment=Remote Desktop
+Exec=rustdesk %u
+Icon=rustdesk
```
Comment (hfiguiere) [inline com.rustdesk.RustDesk.desktop:6]: this icon name is wrong.

File: com.rustdesk.RustDesk.desktop
Comment (hfiguiere) [reply com.rustdesk.RustDesk.desktop:1]: just remove this file.

File: com.rustdesk.RustDesk.json [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+      "name": "rustdesk",
+      "buildsystem": "simple",
+      "build-commands": [
+        "bsdtar -zxvf rustdesk.deb",
+        "tar -xvf ./data.tar.xz",
+        "cp -r usr/* /app/",
+        "mkdir -p /app/bin && ln -s /app/lib/rustdesk/rustdesk /app/bin/rustdesk",
+        "mv /app/share/applications/rustdesk.desktop /app/share/applications/com.rustdesk.RustDesk.desktop",
+        "sed -i '/^Icon=/ c\\Icon=com.rustdesk.RustDesk' /app/share/applications/*.desktop",
+        "mv /app/share/icons/hicolor/scalable/apps/rustdesk.svg /app/share/icons/hicolor/scalable/apps/com.rustdesk.RustDesk.svg",
```
Comment (hfiguiere) [inline com.rustdesk.RustDesk.json:62]: leave the icon alone, remove the sed line (don't use sed anyway), leave the desktop file alone and then use `rename-desktop-file` and `rename-icon` in the manifest to take care of the renaming.

**Comment by besdar** [comment]
Hello everyone!

This is a follow-up to the [previous PR](https://github.com/flathub/flathub/pull/5233). I want to clarify that I am not affiliated with RustDesk in any way; I am just a regular user who would like to see this application available on Flathub.

The previous PR has the following issues:
1) The finish argument `--talk-name=org.freedesktop.Flatpak` poses a significant problem, as it could potentially break out of the sandbox and compromise security. Therefore, it is not permitted without an exception. It seems this argument is necessary to run loginctl ([#1537](https://github.com/rustdesk/rustdesk/discussions/1537)) or to access the desktop when the application is not open. As a regular user who does not operate my own server or a corporate account, I have not been able to log in to verify whether there are any issues with this feature. Additionally, I can only access my PC remotely if the application is opened manually. Everything else I have tested with this manifest works fine for me. As a user, I would be happy to have this application on Flathub even without these features. I am not an expert in Flatpak, so if anyone is willing to address these issues, please feel free to do so by opening a new PR.
2) Regarding the `--filesystem=home` permission, it is necessary because the application has a unique way of allowing both connected users to share files between devices. I hope that the Flathub reviewers can consider making an exception for this permission and simply indicate on the app's Flathub page that this app may not be secure. 
image
If no exceptions can be made, I am willing to remove this permission as well, but that would mean losing a second feature. I believe this functionality could potentially be implemented using portals, a feature that works natively with most toolkits and allows users to select files to send to the application. However, this would require changes to the application's source code, which could be addressed in the future.

Dear reviewers, I welcome any suggestions you may have regarding this PR. I kindly ask that you provide as detailed feedback as possible, as I am new to Flatpak. Thank you for your understanding and support!

File: rustdesk.svg
Comment (hfiguiere) [inline rustdesk.svg:1]: you don't need this apparently as it is in the .deb

File: com.rustdesk.RustDesk.metainfo.xml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+      <li> Supports VP8 / VP9 / AV1 software codecs, and H264 / H265 hardware codecs. </li>
+      <li> Own your data, easily set up self-hosting solution on your infrastructure. </li>
+      <li> P2P connection with end-to-end encryption based on NaCl. </li>
+      <li> No administrative privileges or installation needed for Windows, elevate priviledge locally or from remote on demand. </li>
+      <li> We like to keep things simple and will strive to make simpler where possible. </li>
+    </ul>
+    <p>
+        For self-hosting setup instructions please go to our home page.
+    </p>
+    <p>NOTE: This wrapper is not verified by, affiliated with, or supported by RustDesk.</p>
```
Comment (hfiguiere) [inline com.rustdesk.RustDesk.metainfo.xml:25]: this is only for proprietary applications....

File: com.rustdesk.RustDesk.json (hunk 3 above)
Comment (besdar) [reply com.rustdesk.RustDesk.json:62]: @hfiguiere are you sure about this? I see there is an issue with these properties: https://github.com/flathub/org.flatpak.Builder/issues/71

I am trying to do it too on Ubuntu 24.04 and my `.desktop` and `.svg` file is not renaming.
flatpak-builder 1.4.4

Comment (hfiguiere) [reply com.rustdesk.RustDesk.json:62]: 1/ use the flatpak.
2/ pretty sure, yes.

**Comment by My1** [comment]
Thanks to @besdar for actually making sound arguments, rather than just doing childish arguments like "the others are doing it too".

but yeah regarding the file system permission, to be a bit more specific, the person connecting in has the ability to ask for a file transfer, and when granted gets a 2-pane window similar to total commander or similar tools where one side is the local, and the other the remote machine and allows to transfer files between the systems, in a manner similar to anydesk, if someone has seen that before (I dont know how e.g. Teamviewer does it, as I havent really used it for filetransfers in the past).

Anyway, this method of file transfer doesnt really work well with the portal method of open/save dialogs, especially as the remote person doesnt have to open a screen control session for file transfer in the first place.

File: com.rustdesk.RustDesk.json [hunk 5]
```diff
@@ -0,0 +1,110 @@
+{
+  "id": "com.rustdesk.RustDesk",
+  "runtime": "org.freedesktop.Platform",
+  "runtime-version": "23.08",
```
Comment (bbhtt) [inline com.rustdesk.RustDesk.json:4]:
```suggestion
  "runtime-version": "24.08",
```

24.08 is latest

File: com.rustdesk.RustDesk.json [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+  ],
+  "modules": [
+    "shared-modules/libappindicator/libappindicator-gtk3-12.10.json",
+    {
+      "name": "xdotool",
+      "buildsystem": "simple",
+      "build-commands": [
+        "make -j4 && PREFIX=./build make install",
+        "cp -r ./build/*  /app/"
+      ],
```
Comment (bbhtt) [inline com.rustdesk.RustDesk.json:29]:
```suggestion
      "no-autogen": true,
      "make-install-args": [
        "PREFIX=${FLATPAK_DEST}"
  ],
```

File: com.rustdesk.RustDesk.json [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+          "url": "https://github.com/jordansissel/xdotool/releases/download/v3.20211022.1/xdotool-3.20211022.1.tar.gz",
+          "sha256": "96f0facfde6d78eacad35b91b0f46fecd0b35e474c03e00e30da3fdd345f9ada"
+        }
+      ]
+    },
+    {
+      "name": "pam",    
+      "buildsystem": "simple",
+      "build-commands": [
+          "./configure --disable-selinux --prefix=/app && make -j4 install"
```
Comment (bbhtt) [inline com.rustdesk.RustDesk.json:42]: THis should use autotools buildsystem, you can drop the manual build commands.

File: com.rustdesk.RustDesk.json [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+          "sha256": "eff47a4ecd833fbf18de9686632a70ee8d0794b79aecb217ebd0ce11db4cd0db"
+        }
+      ]
+    },
+    {
+      "name": "rustdesk",
+      "buildsystem": "simple",
+      "build-commands": [
+        "bsdtar -zxvf rustdesk.deb",
+        "tar -xvf ./data.tar.xz",
```
Comment (bbhtt) [inline com.rustdesk.RustDesk.json:57]:
```suggestion
        "bsdtar -Oxf rustdesk.deb data.tar.xz | bsdtar -xf -",
```

File: com.rustdesk.RustDesk.metainfo.xml (hunk 4 above)
Comment (jvschiavo) [reply com.rustdesk.RustDesk.metainfo.xml:25]: Do you mean the line 25?

Comment (bbhtt) [reply com.rustdesk.RustDesk.metainfo.xml:25]: Yes

File: com.rustdesk.RustDesk.desktop (hunk 2 above)
Comment (besdar) [reply com.rustdesk.RustDesk.desktop]: this file was deleted

File: com.rustdesk.RustDesk.json (hunk 7 above)
Comment (besdar) [reply com.rustdesk.RustDesk.json]: done. @bbhtt I hope I have understood you correctly, please check the result. If something is still wrong, please add a code suggestion to the file.

File: com.rustdesk.RustDesk.json [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+        }
+      ]
+    },
+    {
+      "name": "pam",    
+      "buildsystem": "autotools",
+      "config-opts": [ "--disable-selinux" ],
+      "build-commands": [
+          "make -j4 install"
+      ],
```
Comment (bbhtt) [inline com.rustdesk.RustDesk.json:43]:
```suggestion
```

File: com.rustdesk.RustDesk.json (hunk 3 above)
Comment (bbhtt) [reply com.rustdesk.RustDesk.json:61]: can you fix this too ? Just do `rename-desktop-file: rustdesk.desktop` and `rename-icon: rustdesk` on top of the manifest

File: com.rustdesk.RustDesk.metainfo.xml
Comment (bbhtt) [inline com.rustdesk.RustDesk.metainfo.xml:1]: is this metainfo file submitted upstream ?

**Comment by bbhtt** [comment]
https://github.com/flathub/flathub/pull/5670#discussion_r1788042435 needs fixing.

File: com.rustdesk.RustDesk.metainfo.xml
Comment (besdar) [reply com.rustdesk.RustDesk.metainfo.xml:1]: @bbhtt what do you mean?

File: com.rustdesk.RustDesk.json (hunk 3 above)
Comment (besdar) [reply com.rustdesk.RustDesk.json]: done.

@bbhtt thank you! Your comment is really helped me.

File: com.rustdesk.RustDesk.metainfo.xml
Comment (bbhtt) [reply com.rustdesk.RustDesk.metainfo.xml:1]: did you submit it to https://github.com/rustdesk/rustdesk/ ?

**Comment by bbhtt** [comment]
bot, build com.rustdesk.RustDesk

File: com.rustdesk.RustDesk.metainfo.xml
Comment (besdar) [reply com.rustdesk.RustDesk.metainfo.xml:1]: No. Is it a requirement? If so, then I haven't seen it in the flathub manual.

Comment (bbhtt) [reply com.rustdesk.RustDesk.metainfo.xml:1]: Not a hard requirement but it'd be nice if you can open a PR with it.

https://docs.flathub.org/docs/for-app-authors/requirements#required-metadata

> These metadata files should directly come from upstream whenever possible as these are widely adopeted Freedesktop standards that can be used by other distributions as well.

Comment (besdar) [reply com.rustdesk.RustDesk.metainfo.xml:1]: Ok. I'm going to do this anyway with all the fixes from this PR, so let's have an agreement that I'll also add the manifest file when this PR is done?

Comment (bbhtt) [reply com.rustdesk.RustDesk.metainfo.xml:1]: what do you mean by "add manifest file" ? Add it where ?

**Comment by bbhtt** [comment]
Is the above build working ?

File: com.rustdesk.RustDesk.metainfo.xml
Comment (besdar) [reply com.rustdesk.RustDesk.metainfo.xml:1]: @bbhtt sorry. It's a typo. I meant metainfo file.

Comment (bbhtt) [reply com.rustdesk.RustDesk.metainfo.xml:1]: yes that's fine

**Comment by besdar** [comment]
> Is the above build working ?

@bbhtt yes, I've checked it. It is working.

**Comment by xlionjuan** [comment]
Here is still 1.2.4
```xml
  <releases>
    <release version="v1.2.4" date="2024-05-11"/>
  </releases>
```


https://github.com/besdar/flathub/blob/f4d73e54a0be1d0d6ab66ea6224850a3aafb50b2/com.rustdesk.RustDesk.metainfo.xml#L37

**Comment by xlionjuan** [comment]
I can't establish file transfer, remote control works, or this is need to be fixed by upstream?

圖片

**Comment by besdar** [comment]
> I can't establish file transfer, remote control works, or this is need to be fixed by upstream?

@xlionjuan there seems to be something wrong with Rustdesk itself. File transfer connection works from linux (wayland) to any other device, but not from another device to linux. I've checked the flatpak version from the rustdesk repository and it has the same problem.

**Comment by rustdesk** [comment]
> I can't establish file transfer, remote control works, or this is need to be fixed by upstream?
> 
> 圖片

This https://github.com/rustdesk/rustdesk/commit/e7353be0cd0747febada01216a20aeba1ee52ba3 might fix it

**Comment by hfiguiere** [comment]
this is not how you detect you are inside a flatpak. But then the other logic seems to be flawed as well.

**Comment by bbhtt** [comment]
you should check for the existence of `/.flatpak-info` to detect Flatpak sandbox, it's the standard method, not those environment variables.

File: com.rustdesk.RustDesk.metainfo.xml
Comment (besdar) [reply com.rustdesk.RustDesk.metainfo.xml:1]: https://github.com/rustdesk/rustdesk/pull/9581

**Pull request closed without merging**